private/kv_storage.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * (C) Copyright 2023-2025, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR Apache-2.0
+ */
+
+/**
+ * @file kv_storage.h
+ * @brief Wrapper around the NVS library enabling the use of keys longer than 15 chars.
+ *
+ * @details The use of longer keys is made possible by storing both key and value as separate
+ * NVS entries. This leads to overhead in terms of computational complexity and lookup speed.
+ */
+
+#ifndef KV_STORAGE_H
+#define KV_STORAGE_H
+
+#include <esp_err.h>
+#include <nvs.h>
+
+#include "astarte_device_sdk/astarte.h"
+
+typedef struct
+{
+    uint64_t store_index;
+    nvs_handle_t handle;
+    nvs_type_t type;
+} kv_storage_iterator_t;
+
+/**
+ * @brief Sets variable length binary value for given key.
```

Comment:
Do we want to use `3rd`  or `2nd` person? For example here use 3rd https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#styleguide
- `Gets the label.` (preferred)
- `Get the label.` (avoid)

```suggestion
 * @brief Sets? or Set? variable length binary value for given key.
```

Reply:
I would use the second peson. I think it might be more in line with the rest of the documentation.